File: Greedy/Best Time to Buy and Sell Stock II .cs
/*Say you have an array for which the ith element is the price of a given stock on day i.

Design an algorithm to find the maximum profit. You may complete as many transactions as you like (ie, buy one and sell one share of the stock multiple times). However, you may not engage in multiple transactions at the same time (ie, you must sell the stock before you buy again).*/
public class Solution {
    public int MaxProfit(int[] prices) {
       int sum = 0;
       int maxProfit = 0;
       for(int i = 1; i<prices.Length; i++){
       
             if(prices[i] > prices[i-1]){
               sum += prices[i] -  prices[i-1];
             }
            else {
                maxProfit+=sum;
                sum = 0;
            }
       }
       maxProfit+=sum;
       return maxProfit; 
    }
}

File: Best time to buy and sell stock II.cpp
/*
Say you have an array for which the ith element is the price of a given stock on day i.

Design an algorithm to find the maximum profit. You may complete as many transactions as you like (ie, buy one and sell one share of the stock multiple times). However, you may not engage in multiple transactions at the same time (ie, you must sell the stock before you buy again).
*/
class Solution {
public:
    int maxProfit(vector<int> &p) {
        // Start typing your C/C++ solution below
        // DO NOT write int main() function
        int sz = p.size();
        if(sz<=1) return 0;
        int mx=0,sum=0;
        for(int i = 1; i<sz; i++){
            if(p[i]>p[i-1]){
                sum+=p[i] - p[i-1];
            }
            else{
                mx+=sum; sum=0;
            }
        }
        mx+=sum;
        return mx;
    }
};